src/control/XournalMain.cpp

```diff
@@ -557,6 +557,12 @@ void on_startup(GApplication* application, XMPtr app_data) {
             XojMsgBox::showErrorToUser(static_cast<GtkWindow*>(*app_data->win), msg);
             opened = app_data->control->newFile("", p);
         }
+    } else if (app_data->control->getSettings()->isAutoloadMostRecent()) {
+        auto p = Util::fromUri(gtk_recent_info_get_uri(app_data->control->getRecentManager()->getMostRecent()));
+
+        if (p) {
+            opened = app_data->control->openFile(*p);
+        }
```

Comment:
We try to modernize our codebase, so we want new code as modern as possible:
Please use [cpp-17 - init-statements](https://en.cppreference.com/w/cpp/language/if)

```suggestion
        if (auto p = Util::fromUri(gtk_recent_info_get_uri(app_data->control->getRecentManager()->getMostRecent()));
            p) {
            opened = app_data->control->openFile(*p);
        }
```